pumpkin-lib/src/engine/cp/assignments_integer.rs

```diff
@@ -408,11 +415,12 @@ impl AssignmentsInteger {
             let upper_bound_before = self.domains[domain_id].upper_bound;
             let fixed_before = upper_bound_before == lower_bound_before;
 
+            let trail_index = num_trail_entries_before_synchronisation - index - 1;
 
             self.domains[domain_id].undo_trail_entry(&entry);
 
             if fixed_before && self.domains[domain_id].lower_bound != self.domains[domain_id].upper_bound {
-                if is_watching_any_backtrack_events {
+                if is_watching_any_backtrack_events && trail_index < last_notified_trail_index {
```

Comment:
I would put this check in the CSP solver, where we actually notify the propagator. That way this function does not have such an awkward argument.

Reply:
The problem with this is that we need the index of the trail entry which has been synchronised; we could make some assumptions on when the event_drain is extended but I would prefer not to since it could make for some brittle code

Please let me know if you have any suggestions!